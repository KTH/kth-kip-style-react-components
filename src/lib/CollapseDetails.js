import React from 'react'
import PropTypes from 'prop-types'

const CollapseDetails = ({
  ariaLabel = '',
  className = '',
  color = 'white',
  children,
  open = false,
  title = 'Show more information',
  yellow = false,
  ...rest
}) => (
  <details style={yellow ? { backgroundColor: '#fcf8e3' } : {}} className={className} open={open} {...rest}>
    <summary className={color} aria-label={ariaLabel}>
      {title}
    </summary>
    <div
      style={
        yellow
          ? {
              marginLeft: '10px',
              marginRight: '10px',
              paddingBottom: '2px',
            }
          : {}
      }
    >
      {children}
    </div>
  </details>
)

CollapseDetails.propTypes = {
  ariaLabel: PropTypes.string,
  className: PropTypes.string,
  color: PropTypes.oneOf(['blue', 'white']),
  title: PropTypes.string.isRequired,
  open: PropTypes.bool,
  children: PropTypes.node,
}

export default CollapseDetails
