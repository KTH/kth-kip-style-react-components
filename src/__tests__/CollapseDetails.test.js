import React from 'react'
import renderer from 'react-test-renderer'

import CollapseDetails from '../lib/CollapseDetails'

const { getAllByRole } = screen

describe('Component <CollapseDetails>', () => {
  it('renders properly', () => {
    const tree = renderer
      .create(
        <CollapseDetails
          ariaLabel="Expand this to see a helping text"
          className="guidance-per-content"
          title="Show more information"
        >
          <p>Here you can found some text which helps you to fill in all data.</p>
        </CollapseDetails>
      )
      .toJSON()
    console.log(tree)
    expect(tree).toMatchSnapshot()
  })
})
