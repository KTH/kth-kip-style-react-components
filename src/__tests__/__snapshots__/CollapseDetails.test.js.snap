// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <CollapseDetails> renders properly 1`] = `
<details
  className="guidance-per-content"
  open={false}
  style={Object {}}
>
  <summary
    aria-label="Expand this to see a helping text"
    className="white"
  >
    Show more information
  </summary>
  <div
    style={Object {}}
  >
    <p>
      Here you can found some text which helps you to fill in all data.
    </p>
  </div>
</details>
`;
